### PR TITLE
adds template for RDS mediajoint instance

### PR DESCRIPTION
**Pull request by sandikbarr**
intended to be run manually as follows:
`aws cloudformation update-stack --stack-name <stack name> --template-url <template url> --parameters ParameterKey=DBName,ParameterValue=mediajoint_production ParameterKey=DBUser,ParameterValue=<admin but not actually SUPER user> ParameterKey=DBPassword,ParameterValue=<admin user password> ParameterKey=DBAllocatedStorage,ParameterValue=<GB of storage> ParameterKey=DBInstanceClass,ParameterValue=<RDS tier> ParameterKey=EC2SecurityGroup,ParameterValue=<ssh tunnel security group id> ParameterKey=RDSSubnetGroup,ParameterValue=<Platform-Production subnet group name>`

### CODE REVIEW

File: stacks/rds-mediajoint.yml [hunk 1]
```diff
@@ -0,0 +1,124 @@
+AWSTemplateFormatVersion: "2010-09-09"
```
Comment (farski) [inline stacks/rds-mediajoint.yml]: The `/stacks` directory has kind of stumbled into being reserved for stacks that are deployed though CD. I suggest we move this to a `/db` directory (to match `/dns` and `/cdn`, which also have stacks that are managed outside of CD)

File: stacks/rds-mediajoint.yml [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+      StorageType: gp2
+      DBInstanceClass: !Ref DBInstanceClass
+      Engine: MySQL
+      EngineVersion: 5.5.46
+      MasterUsername: !Ref DBUser
+      MasterUserPassword: !Ref DBPassword
+      MultiAZ: !Ref MultiAZ
+      Tags:
+        - Key: Name
+          Value: Master Database
```
Comment (farski) [inline stacks/rds-mediajoint.yml]: There's a number of common tags that should get added to all taggable resources in templates. See _Tag everything_ [here](https://github.com/PRX/Infrastructure/blob/master/cd/README.md) and [an example](https://github.com/PRX/Infrastructure/blob/master/cdn/feed-cdn.yml#L156)

File: stacks/rds-mediajoint.yml [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+      - db.t2.small
+      - db.t2.medium
+      - db.t2.large
+    ConstraintDescription: must select a valid database instance type.
+  EC2SecurityGroup:
+    Description: >-
+      The "MySQL_replication_tunnel" EC2 security group that contains the EC2 instance acting as the replication tunnel
+      https://console.aws.amazon.com/ec2/v2/home?region=us-east-1#SecurityGroups:sort=groupId
+    Default: default
+    Type: String
```
Comment (farski) [inline stacks/rds-mediajoint.yml]: Security group names & ID's and some other AWS related values can be set explicitly as the parameter type https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/parameters-section-structure.html#aws-specific-parameter-types

**Comment by farski** [comment]
I've had better luck with `aws cloudformation deploy` than `aws cloudformation update-stack`. The `deploy` behavior is a lot closer to what the console does, especially in terms of parameters. `update-stack`, I believe, requires all parameters get passed in for every call, where `deploy` will use existing parameter values if a replacement isn't provided.

**Comment by sandikbarr** [comment]
Updated the stack with a `deploy` command. I can see where that's more convenient for small changes like the new tags and parameter type. Back to you @farski

**Pull request merged**